src/DIPS.Xamarin.UI/Controls/RadioButtonGroup/RadioButtonGroup.xaml.cs

```diff
@@ -0,0 +1,323 @@
+        {
+            if (!(bindable is RadioButtonGroup radioButtonGroup))
+                return;
+            if (!(newvalue is Color newColor))
+                return;
+
+            radioButtonGroup.m_radioButtons.ForEach(
+                rb =>
+                {
+                    rb.DeSelectedColor = newColor;
+                    rb.RefreshColor(rb.IsSelected);
+                });
+        }
+
+        /// <summary>
+        ///     Get the value from the <see cref="DisplayMemberPath" />
+        /// </summary>
+        /// <param name="item">The object to look for the value of the property</param>
+        /// <returns></returns>
+        private string GetDisplayPath(object item)
+        {
+            if (string.IsNullOrEmpty(DisplayMemberPath))
+            {
+                return item == null ? string.Empty : item.ToString();
+            }
+
+            if (m_displayMember == null)
+            {
+                m_displayMember = item.GetType().GetProperty(DisplayMemberPath);
+            }
+
+            var value = m_displayMember?.GetValue(item, null);
+            return value == null ? string.Empty : value.ToString();
+        }
+
+        private void Initialize(IEnumerable newItems)
+        {
+            if (newItems == null) return;
+
+            var row = radioButtonContainer.Children.Count;
+
+            //Add separator before the first element
+            if (row == 0)
+            {
+                AddSeparator(row);
+                row++;
+            }
+
+            foreach (var item in newItems)
+            {
+                var radioButton = new RadioButton() { Text = GetDisplayPath(item), Identifier = item };
+                //Initialize radio button
+                radioButton.Initialize(this);
+
+                //Set colors for each radiobutton
+                radioButton.SelectedColor = SelectedColor;
+                radioButton.DeSelectedColor = DeSelectedColor;
+                radioButton.RefreshColor(radioButton.IsSelected);
+
+                //Set margin for each button
+                radioButton.Padding = new Thickness(0, 15, 0, 15);
+
+                //Add each radiobutton to the FlexLayout
+                radioButtonContainer.RowDefinitions.Add(new RowDefinition() { Height = GridLength.Auto });
+                Grid.SetRow(radioButton, row);
+                row++;
+                radioButtonContainer.Children.Add(radioButton);
+                m_radioButtons.Add(radioButton);
+
+                //Add a separator after each radio button
+                AddSeparator(row);
+                row++;
+            }
+        }
+
+        private void AddSeparator(int row)
+        {
+            var boxView = new BoxView() { HeightRequest = 1, BackgroundColor = SeparatorColor, AutomationId = SeparatorAutomationId };
+            radioButtonContainer.RowDefinitions.Add(new RowDefinition() { Height = GridLength.Auto });
+            Grid.SetRow(boxView, row);

```

Comment:
Why use a Grid as the Container? Won't a StackLayout do this much easier?

Reply:
`StackLayout` lead to an issue with sizing when adding items to the `StackLayout` after it was initialized. What happened? The layout did not scale automatically, so each radiobutton tool much more space than it needed.

This was fixed with using a `FlexLayout`, but this layout did not respect its position in the view and took the whole screen.

This is why I ended up with an `Grid`